### PR TITLE
COMMON: Encoding::convertIconv - check errors on last call to `iconv`

### PR DESCRIPTION
**Pull request by ZvikaZ**
There was a bug when trying to convert the Hebrew string `"\e0\e1\e2"`
from WINDOWS-1255 to UTF-32LE, because there was a last call to
`iconv` with `NULL`s, without error checking, and the buffer needed to
be extended for that last one call.

Now this call is inserted to the main `iconv` loop, with error checking.

### CODE REVIEW

File: common/encoding.cpp [hunk 1]
```diff
@@ -244,9 +245,18 @@ char *Encoding::convertIconv(const char *to, const char *from, const char *strin
 				error = true;
 				break;
 			}
+		} else {
+			// we've successfully finished, after the last call with NULLs
+			if (inSize == 0 && src == NULL)
+				loop = false;
+			}
```
Comment (lephilousophe) [inline common/encoding.cpp:252]: There is an indentation problem there

Comment (ZvikaZ) [reply common/encoding.cpp:252]: Fixed

File: common/encoding.cpp [hunk 2]
```diff
@@ -225,7 +225,8 @@ char *Encoding::convertIconv(const char *to, const char *from, const char *strin
 	char *dst = buffer;
 	bool error = false;
 
-	while (inSize > 0) {
+	bool loop = inSize > 0;
+	while (loop) {
 		if (iconv(iconvHandle, &src, &inSize, &dst, &outSize) == ((size_t)-1)) {
```
[the remark this answers is not in the document]
Comment (ZvikaZ) [reply common/encoding.cpp:230]: I understand your other comments, and will fix them. But please check this one again - it is comparing against `(size_t)-1` (if I'm not mistaken ;-) )

[the remark this answers is not in the document]
Comment (lephilousophe) [reply common/encoding.cpp:230]: Ooops. I got to fast on this one. Reviewing code in the GitHub needs me to scroll...
Sorry!

File: common/encoding.cpp [hunk 3]
```diff
@@ -225,7 +225,8 @@ char *Encoding::convertIconv(const char *to, const char *from, const char *strin
 	char *dst = buffer;
 	bool error = false;
 
-	while (inSize > 0) {
+	bool loop = inSize > 0;
+	while (loop) {
```
[the remark this answers is not in the document]
Comment (ZvikaZ) [reply common/encoding.cpp]: Done. Indeed, it's cleaner now.

**Comment by lephilousophe** [comment]
Great, looks good to me.

**Comment by sev-** [comment]
Thanks

**Pull request merged**